### PR TITLE
Public endpoint in service onboarding parity with internal

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements
Handling public endpoints

### CODE REVIEW

File: jazz_create-serverless-service/components/validate-arn.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  @author:
+  @version: 1.0
+**/
+
+var validateARN = (arn) => {
+
+  var arnvalues = arn.split(":");
+  var isarnvalid = true;
+  //validate if arn is a valid Lambda
+  if(arn.indexOf("arn:aws:lambda") !== 0)
```
Comment (svidhani) [inline jazz_create-serverless-service/components/validate-arn.js]: - indentation "if ("
- why not use || and combine the if conditions

File: jazz_create-serverless-service/components/validate-arn.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/** 
+  @module: validate-arn.js
+  @description: Validates ARN value for a lambda function in this case Authorizer
+  @author:
+  @version: 1.0
+**/
+
+var validateARN = (arn) => {
+
+  var arnvalues = arn.split(":");
```
Comment (svidhani) [inline jazz_create-serverless-service/components/validate-arn.js]: what if arn is null or undefined. Have you already validated that? Even if you have for all public(exportable) functions you should check if its not null/undefined.
Please add a unit test for this

File: jazz_create-serverless-service/index.js [hunk 3]
```diff
@@ -75,24 +76,24 @@ module.exports.handler = (event, context, cb) => {
...
-                            } else {
-                                logger.error("Updated service catalog with failed status.");
-                                return cb(JSON.stringify(errorHandler.throwInternalServerError(err.message)));
-                            }
-                        });
+                if (err.jenkins_api_failure) {
+                    serviceDataObject.body = {
+                        "STATUS": "creation_failed"
+                    };
+                    crud.update(serviceId, serviceDataObject, function (serviceUpdateError, results) {
```
Comment (svidhani) [inline jazz_create-serverless-service/index.js]: you can use  (serviceUpdateError, results) => here and other places in the code

File: jazz_create-serverless-service/index.js [hunk 4]
```diff
@@ -221,9 +223,30 @@ module.exports.handler = (event, context, cb) => {
             if (event.body.slack_channel) {
                 inputs.SLACKCHANNEL = event.body.slack_channel;
             }
-            if ((event.body.service_type === "api" || event.body.service_type === "function") && (event.body.require_internal_access !== null)) {
+
+            if ((event.body.service_type === "api" || event.body.service_type === "function") && (event.body.require_internal_access)) {
```
Comment (svidhani) [inline jazz_create-serverless-service/index.js]: you can extract event.body into a variable and re-use it in your code

File: jazz_create-serverless-service/components/validate-arn.js (hunk 1 above)
Comment (SiniWilson) [reply jazz_create-serverless-service/components/validate-arn.js]: Changed with OR

File: jazz_create-serverless-service/components/validate-arn.js (hunk 2 above)
Comment (SiniWilson) [reply jazz_create-serverless-service/components/validate-arn.js]: Already validating the arn. Also included validation in the exportable funciton. Unit test case is handling in the separate story. So leaving the unit test case part as of now.

File: jazz_create-serverless-service/index.js (hunk 3 above)
Comment (SiniWilson) [reply jazz_create-serverless-service/index.js]: changed

File: jazz_create-serverless-service/index.js (hunk 4 above)
Comment (SiniWilson) [reply jazz_create-serverless-service/index.js]: In cooperated the change

**Pull request merged**